### PR TITLE
Add SwSh BerryFarmer which uses sound as feedback

### PR DESCRIPTION
**Pull request by fyex**
Unlike the original Berry Farmer, this program requires you to have a Pokemon in the first party slot which can escape from all tree encounters and OHKO them.
This is why I decided to make it a seperate program. Though, if it is preferred it wouldn't be hard to incorporate it into the original program as an option.

### CODE REVIEW

File: SerialPrograms/Source/PokemonSwSh/Inference/Sounds/PokemonSwSh_BerryTreeRustlingSoundDetector.cpp
Comment (pifopi) [inline SerialPrograms/Source/PokemonSwSh/Inference/Sounds/PokemonSwSh_BerryTreeRustlingSoundDetector.cpp:7]: minor but I think some header could be cleaned up !

File: SerialPrograms/Source/PokemonSwSh/Programs/DateSpamFarmers/PokemonSwSh_DateSpam-BerryFarmer2.cpp
Comment (pifopi) [inline SerialPrograms/Source/PokemonSwSh/Programs/DateSpamFarmers/PokemonSwSh_DateSpam-BerryFarmer2.cpp:101]: could it be declared line 128 directly ?

**Review by pifopi — APPROVED** [review]
Looks good to me ! But I'll let Kuro and / or Gin review this because they're way more knowledgeable on it !

File: SerialPrograms/Source/PokemonSwSh/Inference/Sounds/PokemonSwSh_BerryTreeRustlingSoundDetector.cpp
Comment (Mysticial) [reply SerialPrograms/Source/PokemonSwSh/Inference/Sounds/PokemonSwSh_BerryTreeRustlingSoundDetector.cpp:7]: Yeah, we're in the process of removing Qt from anything that isn't UI code. So you shouldn't need `QString` unless you need some of its library functions which are available elsewhere..

File: SerialPrograms/Source/PokemonSwSh/Programs/DateSpamFarmers/PokemonSwSh_DateSpam-BerryFarmer2.cpp
Comment (Mysticial) [reply SerialPrograms/Source/PokemonSwSh/Programs/DateSpamFarmers/PokemonSwSh_DateSpam-BerryFarmer2.cpp:101]: `Common/Cpp/Time.h` has `WallClock` and `current_time()` if you want to avoid writing the full chrono stuff.

File: SerialPrograms/Source/PokemonSwSh/Programs/DateSpamFarmers/PokemonSwSh_DateSpam-BerryFarmer2.cpp
Comment (Mysticial) [inline SerialPrograms/Source/PokemonSwSh/Programs/DateSpamFarmers/PokemonSwSh_DateSpam-BerryFarmer2.cpp:157]: Is this case rare? Blind waiting for the battle to fully start seems suboptimal? There's a `StandardBattleMenuWatcher` that will wait for it with feedback.

File: SerialPrograms/Source/PokemonSwSh/Programs/DateSpamFarmers/PokemonSwSh_DateSpam-BerryFarmer2.cpp
Comment (Mysticial) [inline SerialPrograms/Source/PokemonSwSh/Programs/DateSpamFarmers/PokemonSwSh_DateSpam-BerryFarmer2.cpp:190]: If I'm reading this right, you're shaking the tree even though it's fast rustling. (thus risking a battle) Is this somehow a calculated optimal algorithm?

File: SerialPrograms/Source/PokemonSwSh/Inference/Sounds/PokemonSwSh_BerryTreeRustlingSoundDetector.cpp
Comment (fyex) [reply SerialPrograms/Source/PokemonSwSh/Inference/Sounds/PokemonSwSh_BerryTreeRustlingSoundDetector.cpp]: fixed

File: SerialPrograms/Source/PokemonSwSh/Programs/DateSpamFarmers/PokemonSwSh_DateSpam-BerryFarmer2.cpp
Comment (fyex) [reply SerialPrograms/Source/PokemonSwSh/Programs/DateSpamFarmers/PokemonSwSh_DateSpam-BerryFarmer2.cpp]: moved declaration to line 128 and changed to WallClock for readability.

File: SerialPrograms/Source/PokemonSwSh/Programs/DateSpamFarmers/PokemonSwSh_DateSpam-BerryFarmer2.cpp
Comment (fyex) [reply SerialPrograms/Source/PokemonSwSh/Programs/DateSpamFarmers/PokemonSwSh_DateSpam-BerryFarmer2.cpp]: A Battle should be quite rare. Changed to `StandardBattleMenuWatcher` anyway since it was already used elswehere.

File: SerialPrograms/Source/PokemonSwSh/Programs/DateSpamFarmers/PokemonSwSh_DateSpam-BerryFarmer2.cpp
Comment (fyex) [reply SerialPrograms/Source/PokemonSwSh/Programs/DateSpamFarmers/PokemonSwSh_DateSpam-BerryFarmer2.cpp:185]: Yes, if there is at least one slow rustling first there will be an additional fetch attempt at the end. While I don't have exact numbers it is quite rare for a battle to happen after the first fast rustling (unless it's right after the very first fetch attempt).
This way should be slightly more efficient if battles are less than one in 20ish extra attempts. 
During my tests battles were less than one in 40.

**Pull request merged**
